tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -234,7 +234,11 @@ def run(self):
                 # Sync cluster global details
                 if "provisioner/%s" % NS.tendrl_context.integration_id \
                     in NS.node_context.tags:
-                    volumes = NS.gluster.objects.Volume().load_all() or []
+                    all_volumes = NS.gluster.objects.Volume().load_all() or []
+                    volumes = []
+                    for volume in all_volumes:
+                        if not eval(volume.deleted):
```

Comment:
The **eval** function lets a python program run python code within itself. It is a security risk to use eval.

Check for volume.delete.lower() == "true"

Reply:
ok i will change this

Reply:
Also only to check the value of `volume.deleted` flag, why we need eval?